File: compiler/engine/testdata/reference/arithmetic/engine.ts
// TypeScript equivalent of arithmetic/engine.go.
import { Archetype, entityMemory, defineImport } from '@sonolus/sonolus.js-compiler/play'

const archetype = new Archetype('N')
const x = entityMemory(Number)
const y = entityMemory(Number)
const z = entityMemory(Number)
defineImport(x)
defineImport(y)
defineImport(z)

export function initialize() {
	x.set(0)
	y.set(0)
	z.set(0)
}

export function updateSequential() {
	x.set((3 + 5) * (2 + 1))
	y.set(x.get() * 2 - 4)
	z.set(x.get() + y.get())
}

File: compiler/engine/testdata/reference/arithmetic/engine.go
package p

type N struct {
	X float64 `sonolus:"memory"`
	Y float64 `sonolus:"memory"`
	Z float64 `sonolus:"memory"`
}

func (n N) Initialize() {
	n.X = 0
	n.Y = 0
	n.Z = 0
}

func (n N) UpdateSequential() {
	n.X = (3 + 5) * (2 + 1)
	n.Y = n.X * 2 - 4
	n.Z = n.X + n.Y
}
